cmd/fpga_plugin/mode.go

```diff
@@ -0,0 +1,56 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+func getModeOverrideFromCluster(nodeName, kubeConfig, master, mode string) (string, error) {
+	var config *rest.Config
+	var err error
+
+	if len(kubeConfig) == 0 {
+		config, err = rest.InClusterConfig()
+	} else {
+		config, err = clientcmd.BuildConfigFromFlags(master, kubeConfig)
+	}
+	if err != nil {
+		return mode, err
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return mode, err
+	}
+
+	if len(nodeName) == 0 {
+		// NODE_NAME is not set so try to use hostname
+		hostName, err := os.Hostname()
```

Comment:
I'm still not sure if this magic is needed. But if @bart0sh needs it I'm fine.

Reply:
I don't need this.

Reply:
this changes the existing behavior and is the reason why this work was done in the first place ;) I can of course drop it

Reply:
Right. Then let's add a deprecation warning and schedule removing it in the next release to avoid problems in existing installations :)

Reply:
dropped. I added a note in the commit message how this is changed.